src/Http/Http.Extensions/gen/StaticRouteHandlerModel/StaticRouteHandlerModel.Emitter.cs

```diff
@@ -211,24 +209,90 @@ private static void EmitCallsToMetadataProvidersForParameters(this Endpoint endp
                 break;
             }
 
+            if (parameter.IsEndpointParameterMetadataProvider)
+            {
+                codeWriter.WriteLine($$"""var {{parameter.SymbolName}}_ParameterInfo = parameterInfos.Single(p => p.Name == "{{parameter.SymbolName}}");""");
```

Comment:
I realized while working on my `AsParameters` PR that you can index directly into the appropriate `ParameterInfo` here by using the `Ordinal` value that we store. AFAIK, `GetParameters` gurantees that the ordering of the `ParameterInfos` will match those in the definition of the method ([ref](https://learn.microsoft.com/en-us/dotnet/api/system.reflection.methodbase.getparameters?view=net-8.0))

```suggestion
                codeWriter.WriteLine($$"""var {{parameter.SymbolName}}_ParameterInfo = parameterInfos[{{parameter.Ordinal}}];""");
```